ambari-server/src/main/java/org/apache/ambari/server/serveraction/kerberos/CreateKeytabFilesServerAction.java

```diff
@@ -217,6 +217,17 @@ protected CommandReport processIdentity(ResolvedKerberosPrincipal resolvedPrinci
                 KerberosPrincipalEntity principalEntity = kerberosPrincipalDAO.find(resolvedPrincipal.getPrincipal());
                 String cachedKeytabPath = (principalEntity == null) ? null : principalEntity.getCachedKeytabPath();
 
+		//This is to hanlde regenerate keytabs for missing hosts if keytab file does not exist in the cache dir.                
+		if(!regenerateKeytabs) {
+                  if(cachedKeytabPath != null) {
+                    File file = new File(cachedKeytabPath);
+                    if(!file.exists()) {
+                      regenerateKeytabs = true;
+                      LOG.info("Regenerating the keytab  since  " + cachedKeytabPath + " is not present");
```

Comment:
please use formatting like `LOG.info("something {} important", importantVar);`

Reply:
sure - will take care of that.